### ISSUE TITLE
Why GPU memory usage keeps ceaselessly growing when Combining my GAN with GN?

### ISSUE THREAD

**Issue by wenhaoyong** [basiclab/GNGAN-PyTorch#2]
After integrating Gradient Normalization into CycleGAN, the GPU memory usage keeps ceaselessly growing, and finally, it blows up.
```
with module_no_grad([netG_A2B]):
    optimizer_D_A.zero_grad()
    pred_real = normalize_gradient(net_D=netD_A, x=real_A)
    pred_fake = normalize_gradient(net_D=netD_A, x=fake_A.detach())
    loss_D_A = -pred_real.mean() + pred_fake.mean() 
    loss_D_A.backward()
    optimizer_D_A.step()
```
I've tried to `del pred_real, pred_fake, loss_D_A` in the code above, even `del grad_norm, f` in gradnorm.py, and it does not help.
The GN sounds great to my work. 
And any tips would be appreciated.

**Comment by w86763777** [comment]
Hi, I didn't encounter the similar problem in usual GAN training. However I can't guarantee that the cycle GAN training is also feasible. Can you provide a minimal reproducable training example with dummy noise as inputs? I will try to figure out the problem.

**Comment by wenhaoyong** [comment]
Hi, I've fixed it. 
This bug exists because the `real_A `in my code is a `torch.autograd.variable`
And we got a line code `x.requires_grad_(True)` in `normalize_gradient` (which in gradnorm.py)
So the gradient of `real_A` will not release without making a backward.
Finally I just `del real_A`, it works and GPU memory usage stays calm down.
